### PR TITLE
(BOLT-266) standardize logging and log actions from plans

### PR DESCRIPTION
**Pull request by adreyer**

### CODE REVIEW

File: lib/bolt/cli.rb [hunk 1]
```diff
@@ -420,8 +420,10 @@ def execute(options)
       executor = Bolt::Executor.new(@config, options[:noop])
 
       if options[:mode] == 'plan'
+        executor = Bolt::Executor.new(@config, options[:noop], true)
```
Comment (MikaelSmith) [inline lib/bolt/cli.rb]: Line 420 seems redundant.

File: lib/bolt/execution_result.rb [hunk 2]
```diff
@@ -47,7 +47,7 @@ def empty
 
     def error_nodes
       result = {}
-      @result_hash.each_pair { |k, v| result[k] = v if v.is_a?(Error) }
+      @result_hash.each_pair { |k, v| result[k] = v if v.is_a?(Puppet::DataTypes::Error) }
```
Comment (MikaelSmith) [inline lib/bolt/execution_result.rb]: Was this needed?

File: lib/bolt/config.rb [hunk 3]
```diff
@@ -1,12 +1,11 @@
-require 'logger'
+require 'bolt/logger'
 require 'yaml'
 require 'bolt/cli'
 
 module Bolt
   Config = Struct.new(
     :concurrency,
     :format,
-    :log_destination,
```
Comment (MikaelSmith) [inline lib/bolt/config.rb]: You removed log_destination here but not log_level, and removed defaults for both later. Seems like log_level could also be removed.

File: lib/bolt/config.rb [hunk 4]
```diff
@@ -187,6 +178,12 @@ def validate
         raise Bolt::CLIError, "Unsupported format: '#{self[:format]}'"
       end
 
+      if self[:transports][:ssh][:sudo_password] && self[:transports][:ssh][:run_as].nil?
+        logger = Logger.get_logger
+        logger.warn("'--sudo-password will not be used without specifying a " \
```
Comment (MikaelSmith) [inline lib/bolt/config.rb]: Should `--sudo-password` be wrapped in single-quotes? There's a starting single-quote but not ending.

File: modules/boltlib/lib/puppet/functions/file_upload.rb [hunk 5]
```diff
@@ -18,6 +18,15 @@
     return_type 'ExecutionResult'
   end
 
+  def log_summary(object, node_count, fail_count)
```
Comment (MikaelSmith) [inline modules/boltlib/lib/puppet/functions/file_upload.rb]: This seems to be unused.

File: lib/bolt/execution_result.rb (hunk 2 above)
Comment (adreyer) [reply lib/bolt/execution_result.rb]: yes it fixed an unreported bug. I moved it to a separate commit. I'll write a test too

File: lib/bolt/config.rb (hunk 3 above)
Comment (adreyer) [reply lib/bolt/config.rb]: No the default is handled elsewhere but it's still configurable. We never allowed log_destination to be configured.

File: lib/bolt/config.rb (hunk 4 above)
Comment (adreyer) [reply lib/bolt/config.rb]: we don't do that a few lines below so I'll  just remove it

File: modules/boltlib/lib/puppet/functions/file_upload.rb (hunk 5 above)
Comment (adreyer) [reply modules/boltlib/lib/puppet/functions/file_upload.rb]: yup

File: lib/bolt/cli.rb (hunk 1 above)
Comment (adreyer) [reply lib/bolt/cli.rb]: yup

File: lib/bolt/cli.rb [hunk 6]
```diff
@@ -420,8 +420,10 @@ def execute(options)
       executor = Bolt::Executor.new(@config, options[:noop])
```
Comment (MikaelSmith) [inline lib/bolt/cli.rb]: Still here.

File: lib/bolt/config.rb (hunk 3 above)
Comment (adreyer) [reply lib/bolt/config.rb]: We still configure loglevel we just manage the default in the logger. log_destination is not configured yet and never has been.

File: lib/bolt/config.rb (hunk 4 above)
Comment (adreyer) [reply lib/bolt/config.rb]: we don't quote a flag a few lines down so I removed it

**Pull request merged**